src/ranges.jl

```diff
@@ -90,6 +90,10 @@ struct OptionallyStaticUnitRange{T,F,L} <: AbstractUnitRange{T}
   end
 end
 
+Base.:(:)(L, ::Static{U}) where {U} = OptionallyStaticUnitRange(L, Val(U))
```

Comment:
Yes, I'll make that change and clean up the implementation a little.
I think it should be okay for `first` and `last` to also return `Static`, so that their definitions can be simplified to `r.start` and `r.stop`, respectively.
Then if `r` is an `OptionallyStaticUnitRange`, we should have that `r === first(r):last(r)`.

Reply:
Yes, this makes sense.